value.go

```diff
@@ -351,22 +350,21 @@ func (r *Value) nth(n int) zcode.Bytes {
 	return zv
 }
 
-func (r *Value) Fields() []Field {
+func (r Value) Fields() []Field {
 	return TypeRecordOf(r.Type()).Fields
 }
 
 func (v *Value) DerefByColumn(col int) *Value {
```

Comment:
Maybe this should just return a non-pointer Value and bool if it exists?

Reply:
Quite possibly, though then we'll lose the ability to do method chaining, as in val.DerefByColumn(1).DerefByColumn(2). But given the size of this change so far, I think it's something we should address later. Sound OK?